packages/keystone/authStrategy/passwordAuth.js

```diff
@@ -0,0 +1,131 @@
+const { PasswordAuthStrategy: DefaultPasswordAuthStrategy } = require('@keystonejs/auth-password')
+
+/**
+ * This is a custom authentication strategy specific to condo applications
+ *
+ * Based on PasswordAuthStrategy from "@keystonejs/auth-password"
+ *
+ * What are the differences with the main version?
+ *
+ * - User type check (disabled by default)
+ * - Verification check (disabled by default)
+ * - Reject deleted user authorization (enabled by default)
+ *
+ * All custom features can be turned off, then there will be default authorization like in basic PasswordAuthStrategy
+ *
+ * These features are configurable via "config"
+ *
+ * @example config example
+ * config = {
+ *     // defaults props from "@keystonejs/auth-password"
+ *     identityField: 'email',
+ *     secretField: 'password',
+ *     protectIdentities: true,
+ *
+ *     // custom props
+ *     itemTypeField: 'type',
+ *     itemTypeOptions: ['staff', 'service', 'resident'],
+ *     itemTypeDefault: 'staff',
+ *
+ *     verificationField: 'isEmailVerified',
+ *
+ *     rejectSoftDeletedItems: true,
+ * }
+ */
+class ExtendedPasswordAuthStrategy extends DefaultPasswordAuthStrategy {
+    constructor (...props) {
+        super(...props)
+
+        this.config = {
+            // NOTE: We do not set default values for new fields as the implementation of our applications may differ.
+            // We should configure each application yourself as we need.
+            //
+            // itemTypeField: 'type',
+            // itemTypeOptions: ['staff', 'service', 'resident'],
+            // itemTypeDefault: 'staff',
+            //
+            // verificationField: 'isEmailVerified',
+
+            rejectSoftDeletedItems: true,
+
+            ...this.config,
+        }
+    }
+
+    getInputFragment () {
+        const itemTypeField = this.config.itemTypeField
+            ? `${this.config.itemTypeField}: String`
+            : ''
+
+        return `
+          ${this.config.identityField}: String
+          ${this.config.secretField}: String
+          ${itemTypeField}
+        `
+    }
+
+    async validate (args) {
+        const { itemTypeField, itemTypeOptions, itemTypeDefault } = this.config
+        const itemType = args[itemTypeField] || itemTypeDefault || null
+
+        if (Boolean(itemTypeField) && Array.isArray(itemTypeOptions) && !itemTypeOptions.includes(itemType)) {
+            const message = `[passwordAuth:itemType:invalid] The ${itemTypeField} contains an invalid value. It must be one of the following: ${itemTypeOptions.join(', ')}`
+            return { success: false, message }
```

Comment:
Is this API dictated by Keystone? We have a lot of ways to say that function failed, lets use smth that is used elsewhere. Like isOk in notifications domain

Reply:
This is the implementation like in keystone, so I would stick to the same API